--- decompiled/TerrariaServerWindows/Terraria/GameContent/UI/WiresUI.cs
+++ decompiled/TerrariaServerLinux/Terraria/GameContent/UI/WiresUI.cs
@@ -315,11 +_,11 @@
 						if (i == 3) {
 							num5 = 0f;
 						}
-						vector4 = vector2 + Vector2.UnitX.RotatedBy(num5 * ((float)Math.PI * 2f) / num3 - (float)Math.PI / num4) * 45f;
+						vector4 = vector2 + Vector2.UnitX.RotatedBy(num5 * (MathF.PI * 2f) / num3 - MathF.PI / num4) * 45f;
 						break;
 					}
 					case 5:
-						vector4 = vector2 + Vector2.UnitX.RotatedBy((float)(i - 1) * ((float)Math.PI * 2f) / num3 - (float)Math.PI / num4) * 45f;
+						vector4 = vector2 + Vector2.UnitX.RotatedBy((float)(i - 1) * (MathF.PI * 2f) / num3 - MathF.PI / num4) * 45f;
 						break;
 					case 4:
 						flag2 = false;
@@ -339,7 +_,7 @@
 					case 2:
 					case 3:
 					case 5: {
-						float value = (vector4 - vector2).ToRotation().AngleTowards(num, (float)Math.PI * 2f / (num3 * 2f)) - num;
+						float value = (vector4 - vector2).ToRotation().AngleTowards(num, MathF.PI * 2f / (num3 * 2f)) - num;
 						if (num2 >= 20f && Math.Abs(value) < 0.01f) {
 							flag3 = true;
 						}
