--- decompiled/TerrariaServerWindows/Terraria/UI/FancyErrorPrinter.cs
+++ decompiled/TerrariaServerLinux/Terraria/UI/FancyErrorPrinter.cs
@@ -131,7 +_,7 @@
 	{
 		if (Platform.IsWindows && !Main.dedServ)
 		{
-			MessageBox.Show(preparedMessage, "Error", MessageBoxButtons.OK, MessageBoxIcon.Hand);
+			MessageBox.Show(preparedMessage, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
 		}
 	}
 }
