--- decompiled/TerrariaClientWindows/Terraria/IO/PlayerFileData.cs
+++ decompiled/TerrariaClientLinux/Terraria/IO/PlayerFileData.cs
@@ -74,7 +_,11 @@
 			return;
 		}
 		string fileName = GetFileName(includeExtension: false);
-		string path = Main.PlayerPath + System.IO.Path.DirectorySeparatorChar + fileName + System.IO.Path.DirectorySeparatorChar;
+		string playerPath = Main.PlayerPath;
+		char directorySeparatorChar = System.IO.Path.DirectorySeparatorChar;
+		string text = directorySeparatorChar.ToString();
+		directorySeparatorChar = System.IO.Path.DirectorySeparatorChar;
+		string path = playerPath + text + fileName + directorySeparatorChar;
 		if (Directory.Exists(path)) {
 			string[] files = Directory.GetFiles(path);
 			for (int i = 0; i < files.Length; i++) {
