--- decompiled/TerrariaServerWindows/Terraria/GameContent/PlayerSleepingHelper.cs
+++ decompiled/TerrariaServerLinux/Terraria/GameContent/PlayerSleepingHelper.cs
@@ -70,7 +_,7 @@
 			isSleeping = state;
 			if (state)
 			{
-				player.fullRotation = (float)Math.PI / 2f * (float)(-player.direction);
+				player.fullRotation = MathF.PI / 2f * (float)(-player.direction);
 				return;
 			}
 			player.fullRotation = 0f;
