--- decompiled/TerrariaServerWindows/Terraria/Graphics/Capture/CaptureCamera.cs
+++ decompiled/TerrariaServerMac/Terraria/Graphics/Capture/CaptureCamera.cs
@@ -220,7 +_,11 @@
 
 	private bool SaveImage(int width, int height, ImageFormat imageFormat, string filename)
 	{
-		if (!Utils.TryCreatingDirectory(Main.SavePath + Path.DirectorySeparatorChar + "Captures" + Path.DirectorySeparatorChar)) {
+		string savePath = Main.SavePath;
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
+		string text = directorySeparatorChar.ToString();
+		directorySeparatorChar = Path.DirectorySeparatorChar;
+		if (!Utils.TryCreatingDirectory(savePath + text + "Captures" + directorySeparatorChar)) {
 			return false;
 		}
 		try {
@@ -246,7 +_,20 @@
 
 	private void SaveImage(Texture2D texture, int width, int height, ImageFormat imageFormat, string foldername, string filename)
 	{
-		string text = Main.SavePath + Path.DirectorySeparatorChar + "Captures" + Path.DirectorySeparatorChar + foldername;
+		string[] obj = new string[5] {
+			Main.SavePath,
+			null,
+			null,
+			null,
+			null
+		};
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
+		obj[1] = directorySeparatorChar.ToString();
+		obj[2] = "Captures";
+		directorySeparatorChar = Path.DirectorySeparatorChar;
+		obj[3] = directorySeparatorChar.ToString();
+		obj[4] = foldername;
+		string text = string.Concat(obj);
 		string text2 = Path.Combine(text, filename);
 		if (!Utils.TryCreatingDirectory(text)) {
 			return;
@@ -300,7 +_,28 @@
 	{
 		if (_activeSettings.UseScaling) {
 			int num = 0;
-			while (!SaveImage(_outputImageSize.Width, _outputImageSize.Height, ImageFormat.Png, Main.SavePath + Path.DirectorySeparatorChar + "Captures" + Path.DirectorySeparatorChar + _activeSettings.OutputName + ".png")) {
+			while (true) {
+				int width = _outputImageSize.Width;
+				int height = _outputImageSize.Height;
+				ImageFormat png = ImageFormat.Png;
+				string[] obj = new string[6] {
+					Main.SavePath,
+					null,
+					null,
+					null,
+					null,
+					null
+				};
+				char directorySeparatorChar = Path.DirectorySeparatorChar;
+				obj[1] = directorySeparatorChar.ToString();
+				obj[2] = "Captures";
+				directorySeparatorChar = Path.DirectorySeparatorChar;
+				obj[3] = directorySeparatorChar.ToString();
+				obj[4] = _activeSettings.OutputName;
+				obj[5] = ".png";
+				if (SaveImage(width, height, png, string.Concat(obj))) {
+					break;
+				}
 				GC.Collect();
 				Thread.Sleep(5);
 				num++;
