--- decompiled/TerrariaServerWindows/Terraria/GameContent/Golf/FancyGolfPredictionLine.cs
+++ decompiled/TerrariaServerLinux/Terraria/GameContent/Golf/FancyGolfPredictionLine.cs
@@ -83,7 +_,7 @@
 	{
 		float num = travelledLength % 200f / 200f;
 		num *= (float)_colors.Length;
-		num -= _time * (float)Math.PI * 1.5f;
+		num -= _time * MathF.PI * 1.5f;
 		num %= (float)_colors.Length;
 		if (num < 0f) {
 			num += (float)_colors.Length;
@@ -100,7 +_,7 @@
 
 	private Color GetColor2(float index)
 	{
-		float num = index * 0.5f - _time * (float)Math.PI * 1.5f;
+		float num = index * 0.5f - _time * MathF.PI * 1.5f;
 		int num2 = (int)Math.Floor(num) % _colors.Length;
 		if (num2 < 0) {
 			num2 += _colors.Length;
@@ -114,7 +_,7 @@
 
 	private float GetScale(float travelledLength)
 	{
-		return 0.2f + Utils.GetLerpValue(0.8f, 1f, (float)Math.Cos(travelledLength / 50f + _time * -(float)Math.PI) * 0.5f + 0.5f, clamped: true) * 0.15f;
+		return 0.2f + Utils.GetLerpValue(0.8f, 1f, (float)Math.Cos(travelledLength / 50f + _time * -MathF.PI) * 0.5f + 0.5f, clamped: true) * 0.15f;
 	}
 
 	private void GetSectionLength(int startIndex, out float length, out float rotation)
