--- decompiled/TerrariaServerWindows/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs
+++ decompiled/TerrariaServerLinux/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs
@@ -273,6 +_,7 @@
 		{
 			return 1;
 		}
+		int id = _achievement.Id;
-		return _achievement.Id.CompareTo(uIAchievementListItem._achievement.Id);
+		return id.CompareTo(uIAchievementListItem._achievement.Id);
 	}
 }
