--- decompiled/TerrariaServerWindows/Terraria/GameContent/PotionOfReturnGateHelper.cs
+++ decompiled/TerrariaServerMac/Terraria/GameContent/PotionOfReturnGateHelper.cs
@@ -118,9 +_,9 @@
 			DrawData item = drawData;
 			item.color = new Color(127, 50, 127, 0) * _opacity;
 			item.scale *= 1.1f;
-			float x = (Main.GlobalTimeWrappedHourly / 5f * ((float)Math.PI * 2f)).ToRotationVector2().X;
+			float x = (Main.GlobalTimeWrappedHourly / 5f * (MathF.PI * 2f)).ToRotationVector2().X;
 			item.color *= x * 0.1f + 0.3f;
-			item.position += ((Main.GlobalTimeWrappedHourly / 5f + num2) * ((float)Math.PI * 2f)).ToRotationVector2() * (x * 1f + 2f);
+			item.position += ((Main.GlobalTimeWrappedHourly / 5f + num2) * (MathF.PI * 2f)).ToRotationVector2() * (x * 1f + 2f);
 			drawDataList.Add(item);
 		}
 		if (selectionMode != 0) {
