--- decompiled/TerrariaClientWindows/Terraria/Audio/LegacySoundPlayer.cs
+++ decompiled/TerrariaClientLinux/Terraria/Audio/LegacySoundPlayer.cs
@@ -243,7 +_,10 @@
 		TrackableSounds = new Asset<SoundEffect>[SoundID.TrackableLegacySoundCount];
 		TrackableSoundInstances = new SoundEffectInstance[TrackableSounds.Length];
 		for (int num4 = 0; num4 < TrackableSounds.Length; num4++) {
+			Asset<SoundEffect>[] trackableSounds = TrackableSounds;
+			int num5 = num4;
+			char directorySeparatorChar = Path.DirectorySeparatorChar;
-			TrackableSounds[num4] = Load("Sounds/Custom" + Path.DirectorySeparatorChar + SoundID.GetTrackableLegacySoundPath(num4));
+			trackableSounds[num5] = Load("Sounds/Custom" + directorySeparatorChar + SoundID.GetTrackableLegacySoundPath(num4));
 		}
 	}
 
