--- decompiled/TerrariaServerWindows/Terraria/Collision.cs
+++ decompiled/TerrariaServerMac/Terraria/Collision.cs
@@ -33,7 +_,7 @@
 
 	public static bool down;
 
-	public static float Epsilon = (float)Math.E;
+	public static float Epsilon = MathF.E;
 
 	private static List<Point> _cacheForConveyorBelts = new List<Point>();
 
@@ -2621,8 +_,8 @@
 					if (tile.type != 19) {
 						return num5 switch {
 							1 => 0f, 
-							2 => (float)Math.PI / 4f, 
+							2 => MathF.PI / 4f, 
-							3 => -(float)Math.PI / 4f, 
+							3 => -MathF.PI / 4f, 
 							_ => 0f, 
 						};
 					}
@@ -2635,20 +_,20 @@
 						case 19:
 						case 21:
 						case 23:
-							return -(float)Math.PI / 4f;
+							return -MathF.PI / 4f;
 						case 10:
 						case 20:
 						case 22:
 						case 24:
-							return (float)Math.PI / 4f;
+							return MathF.PI / 4f;
 						case 25:
 						case 26:
 							if (!flag) {
 								switch (num5) {
 									case 2:
-										return (float)Math.PI / 4f;
+										return MathF.PI / 4f;
 									case 3:
-										return -(float)Math.PI / 4f;
+										return -MathF.PI / 4f;
 								}
 								break;
 							}
