--- decompiled/TerrariaServerWindows/Terraria/TimeLogger.cs
+++ decompiled/TerrariaServerMac/Terraria/TimeLogger.cs
@@ -142,7 +_,9 @@
 		if (startLoggingNextFrame) {
 			startLoggingNextFrame = false;
 			_ = DateTime.Now;
+			string savePath = Main.SavePath;
+			char directorySeparatorChar = Path.DirectorySeparatorChar;
-			string path = Main.SavePath + Path.DirectorySeparatorChar + "TerrariaDrawLog.7z";
+			string path = savePath + directorySeparatorChar + "TerrariaDrawLog.7z";
 			try {
 				logWriter = new StreamWriter(new GZipStream(new FileStream(path, FileMode.Create), CompressionMode.Compress));
 				logBuilder = new StringBuilder(5000);
