--- decompiled/TerrariaServerWindows/Terraria/Achievements/AchievementManager.cs
+++ decompiled/TerrariaServerLinux/Terraria/Achievements/AchievementManager.cs
@@ -44,7 +_,9 @@
 			_isCloudSave = true;
 			_cryptoKey = SocialAPI.Achievements.GetEncryptionKey();
 		} else {
-			_savePath = Main.SavePath + Path.DirectorySeparatorChar + "achievements.dat";
+			string savePath = Main.SavePath;
+			char directorySeparatorChar = Path.DirectorySeparatorChar;
+			_savePath = savePath + directorySeparatorChar + "achievements.dat";
 			_isCloudSave = false;
 			_cryptoKey = Encoding.ASCII.GetBytes("RELOGIC-TERRARIA");
 		}
