--- decompiled/TerrariaServerWindows/Terraria/Social/Steam/WorkshopSocialModule.cs
+++ decompiled/TerrariaServerLinux/Terraria/Social/Steam/WorkshopSocialModule.cs
@@ -27,7 +_,9 @@
 		_publisherInstances = new List<WorkshopHelper.UGCBased.APublisherInstance>();
 		base.ProgressReporter = new WorkshopProgressReporter(_publisherInstances);
 		base.SupportedTags = new SupportedWorkshopTags();
-		_contentBaseFolder = Main.SavePath + Path.DirectorySeparatorChar + "Workshop";
+		string savePath = Main.SavePath;
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
+		_contentBaseFolder = savePath + directorySeparatorChar + "Workshop";
 		_downloader = WorkshopHelper.UGCBased.Downloader.Create();
 		_publishedItems = WorkshopHelper.UGCBased.PublishedItemsFinder.Create();
 		WorkshopIssueReporter workshopIssueReporter = new WorkshopIssueReporter();
@@ -65,7 +_,10 @@
 
 	public override List<string> GetListOfSubscribedWorldPaths()
 	{
-		return _downloader.WorldPaths.Select((string folderPath) => folderPath + Path.DirectorySeparatorChar + "world.wld").ToList();
+		return _downloader.WorldPaths.Select(delegate(string folderPath) {
+			char directorySeparatorChar = Path.DirectorySeparatorChar;
+			return folderPath + directorySeparatorChar + "world.wld";
+		}).ToList();
 	}
 
 	public override List<string> GetListOfSubscribedResourcePackPaths()
@@ -151,7 +_,12 @@
 	private string GetTemporaryFolderPath()
 	{
 		ulong steamID = SteamUser.GetSteamID().m_SteamID;
-		return _contentBaseFolder + Path.DirectorySeparatorChar + steamID + Path.DirectorySeparatorChar;
+		string contentBaseFolder = _contentBaseFolder;
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
+		string text = directorySeparatorChar.ToString();
+		string text2 = steamID.ToString();
+		directorySeparatorChar = Path.DirectorySeparatorChar;
+		return contentBaseFolder + text + text2 + directorySeparatorChar;
 	}
 
 	private bool MakeTemporaryFolder(string temporaryFolderPath)
@@ -186,7 +_,8 @@
 		if (!Directory.Exists(text)) {
 			return false;
 		}
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
-		if (AWorkshopEntry.TryReadingManifest(text + Path.DirectorySeparatorChar + "workshop.json", out info)) {
+		if (AWorkshopEntry.TryReadingManifest(text + directorySeparatorChar + "workshop.json", out info)) {
 			return true;
 		}
 		return false;
@@ -199,7 +_,8 @@
 		if (!Directory.Exists(fullPath)) {
 			return false;
 		}
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
-		if (AWorkshopEntry.TryReadingManifest(fullPath + Path.DirectorySeparatorChar + "workshop.json", out info)) {
+		if (AWorkshopEntry.TryReadingManifest(fullPath + directorySeparatorChar + "workshop.json", out info)) {
 			return true;
 		}
 		return false;
