--- decompiled/TerrariaServerWindows/Terraria/Map/MapHelper.cs
+++ decompiled/TerrariaServerMac/Terraria/Map/MapHelper.cs
@@ -2913,7 +_,9 @@
 		{
 			Utils.TryCreatingDirectory(text);
 		}
-		text += Path.DirectorySeparatorChar;
+		string text2 = text;
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
+		text = text2 + directorySeparatorChar;
 		text = ((!Main.ActiveWorldFileData.UseGuidAsMapName) ? (text + Main.worldID + ".map") : string.Concat(text, Main.ActiveWorldFileData.UniqueId, ".map"));
 		new Stopwatch().Start();
 		if (!Main.gameMenu)
