--- decompiled/TerrariaServerWindows/ReLogic/Content/Sources/FileSystemContentSource.cs
+++ decompiled/TerrariaServerMac/ReLogic/Content/Sources/FileSystemContentSource.cs
@@ -21,7 +_,9 @@
 		_basePath = Path.GetFullPath(basePath);
 		if (!_basePath.EndsWith("/") && !_basePath.EndsWith("\\"))
 		{
-			_basePath += Path.DirectorySeparatorChar;
+			string basePath2 = _basePath;
+			char directorySeparatorChar = Path.DirectorySeparatorChar;
+			_basePath = basePath2 + directorySeparatorChar;
 		}
 		BuildNameToAbsolutePathDictionary();
 	}
