--- decompiled/TerrariaClientWindows/Terraria/Dust.cs
+++ decompiled/TerrariaClientLinux/Terraria/Dust.cs
@@ -2557,7 +_,7 @@
 					num6 = (int)((float)num6 * num7);
 					num5 = (int)((float)num5 * num7);
 					num4 = (int)((float)num4 * num7);
-					int a2 = (int)(100f * num7);
+					int num8 = (int)(100f * num7);
 					num6 += 50;
 					if (num6 > 255)
 					{
@@ -2573,7 +_,7 @@
 					{
 						num4 = 255;
 					}
-					return new Color(num6, num5, num4, a2);
+					return new Color(num6, num5, num4, num8);
 				}
 				if (type == 284)
 				{
@@ -2615,16 +_,16 @@
 				num6 = (int)((float)(int)newColor.R * num);
 				num5 = (int)((float)(int)newColor.G * num);
 				num4 = (int)((float)(int)newColor.B * num);
-				int num8 = newColor.A - alpha;
+				int num9 = newColor.A - alpha;
-				if (num8 < 0)
+				if (num9 < 0)
 				{
-					num8 = 0;
+					num9 = 0;
 				}
-				if (num8 > 255)
+				if (num9 > 255)
 				{
-					num8 = 255;
+					num9 = 255;
 				}
-				return new Color(num6, num5, num4, num8);
+				return new Color(num6, num5, num4, num9);
 			}
 		}
 	}
