--- decompiled/TerrariaClientWindows/Terraria/Minecart.cs
+++ decompiled/TerrariaClientLinux/Terraria/Minecart.cs
@@ -680,7 +_,7 @@
 						if (trackState == TrackState.OnFront)
 						{
 							trackState = TrackState.NoTrack;
-							vector5 = Vector2.Transform(matrix: (Velocity.X > 0f) ? ((_leftSideConnection[num9] != 2) ? Matrix.CreateRotationZ((float)Math.PI / 4f) : Matrix.CreateRotationZ(-(float)Math.PI / 4f)) : ((_rightSideConnection[num9] != 2) ? Matrix.CreateRotationZ(-(float)Math.PI / 4f) : Matrix.CreateRotationZ((float)Math.PI / 4f)), position: new Vector2(Velocity.X, 0f));
+							vector5 = Vector2.Transform(matrix: (Velocity.X > 0f) ? ((_leftSideConnection[num9] != 2) ? Matrix.CreateRotationZ(MathF.PI / 4f) : Matrix.CreateRotationZ(-MathF.PI / 4f)) : ((_rightSideConnection[num9] != 2) ? Matrix.CreateRotationZ(-MathF.PI / 4f) : Matrix.CreateRotationZ(MathF.PI / 4f)), position: new Vector2(Velocity.X, 0f));
 							vector5.X = Velocity.X;
 							flag4 = true;
 							num = 0f;
