--- decompiled/TerrariaServerWindows/Terraria/GameContent/UI/States/UIWorldCreation.cs
+++ decompiled/TerrariaServerLinux/Terraria/GameContent/UI/States/UIWorldCreation.cs
@@ -701,7 +_,7 @@
 	{
 		processedSeed = _optionSeed;
 		ProcessSpecialWorldSeeds(processedSeed);
-		string[] array = _optionSeed.Split('.');
+		string[] array = _optionSeed.Split(new char[1] { '.' });
 		if (array.Length != 4) {
 			return;
 		}
