--- decompiled/TerrariaBuildable/ReLogic/OS/Windows/PathService.cs
+++ decompiled/TerrariaUnified/ReLogic/OS/Windows/PathService.cs
@@ -9,6 +_,8 @@
 {
 	public override string GetStoragePath()
 	{
+		// TODO: Unix builds change Personal to MyDocuments, but this code isn't
+		// reachable on those platforms. Is this change necessary?
 		return Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), "My Games");
 	}
 
