--- decompiled/TerrariaBuildable/ReLogic/OS/Windows/WindowService.cs
+++ decompiled/TerrariaUnified/ReLogic/OS/Windows/WindowService.cs
@@ -10,7 +_,7 @@
 	public float GetScaling()
 	{
 		try {
-			IntPtr hdc = System.Drawing.Graphics.FromHwnd(IntPtr.Zero).GetHdc();
+			IntPtr hdc = NativeMethods.GetDC(IntPtr.Zero);
 			int deviceCaps = NativeMethods.GetDeviceCaps(hdc, NativeMethods.DeviceCap.VertRes);
 			return (float)NativeMethods.GetDeviceCaps(hdc, NativeMethods.DeviceCap.DesktopVertRes) / (float)deviceCaps;
 		} catch (Exception) {
