--- decompiled/TerrariaClientWindows/Terraria/IO/ResourcePackContentValidator.cs
+++ decompiled/TerrariaClientMac/Terraria/IO/ResourcePackContentValidator.cs
@@ -1,5 +_,6 @@
 using System.IO;
 using ReLogic.Content;
+using ReLogic.Content.Sources;
 using Terraria.GameContent;
 
 namespace Terraria.IO;
@@ -10,7 +_,9 @@
 	{
 		if ((AssetReaderCollection)Main.instance.Services.GetService(typeof(AssetReaderCollection)) != null)
 		{
-			pack.GetContentSource().GetAllAssetsStartingWith("Images" + Path.DirectorySeparatorChar);
+			IContentSource contentSource = pack.GetContentSource();
+			char directorySeparatorChar = Path.DirectorySeparatorChar;
+			contentSource.GetAllAssetsStartingWith("Images" + directorySeparatorChar);
 			VanillaContentValidator.Instance.GetValidImageFilePaths();
 		}
 	}
