--- decompiled/TerrariaServerWindows/ReLogic/Peripherals/RGB/RgbKey.cs
+++ decompiled/TerrariaServerMac/ReLogic/Peripherals/RGB/RgbKey.cs
@@ -126,7 +_,7 @@
 
 	private void UpdateFlashingEffect()
 	{
-		float amount = (float)Math.Sin(_timeRemaining * _effectRate * ((float)Math.PI * 2f)) * 0.5f + 0.5f;
+		float amount = (float)Math.Sin(_timeRemaining * _effectRate * (MathF.PI * 2f)) * 0.5f + 0.5f;
 		CurrentColor = Color.Lerp(BaseColor, TargetColor, amount);
 	}
 }
