--- decompiled/TerrariaClientWindows/Terraria/Achievements/AchievementManager.cs
+++ decompiled/TerrariaClientMac/Terraria/Achievements/AchievementManager.cs
@@ -47,7 +_,9 @@
 		}
 		else
 		{
-			_savePath = Main.SavePath + Path.DirectorySeparatorChar + "achievements.dat";
+			string savePath = Main.SavePath;
+			char directorySeparatorChar = Path.DirectorySeparatorChar;
+			_savePath = savePath + directorySeparatorChar + "achievements.dat";
 			_isCloudSave = false;
 			_cryptoKey = Encoding.ASCII.GetBytes("RELOGIC-TERRARIA");
 		}
