--- decompiled/TerrariaClientWindows/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs
+++ decompiled/TerrariaClientMac/Terraria/GameContent/UI/Elements/UIAchievementListItem.cs
@@ -250,6 +_,7 @@
 		if (!_achievement.IsCompleted && uIAchievementListItem._achievement.IsCompleted) {
 			return 1;
 		}
+		int id = _achievement.Id;
-		return _achievement.Id.CompareTo(uIAchievementListItem._achievement.Id);
+		return id.CompareTo(uIAchievementListItem._achievement.Id);
 	}
 }
