--- decompiled/TerrariaServerWindows/Terraria/ID/NPCID.cs
+++ decompiled/TerrariaServerLinux/Terraria/ID/NPCID.cs
@@ -4725,7 +_,7 @@
 					43,
 					new NPCBestiaryDrawModifiers(0) {
 						Position = new Vector2(4f, -6f),
-						Rotation = (float)Math.PI * 3f / 4f
+						Rotation = MathF.PI * 3f / 4f
 					}
 				},
 				{
@@ -4805,7 +_,7 @@
 					56,
 					new NPCBestiaryDrawModifiers(0) {
 						Position = new Vector2(4f, -6f),
-						Rotation = (float)Math.PI * 3f / 4f
+						Rotation = MathF.PI * 3f / 4f
 					}
 				},
 				{
@@ -5056,7 +_,7 @@
 					101,
 					new NPCBestiaryDrawModifiers(0) {
 						Position = new Vector2(-2f, 6f),
-						Rotation = (float)Math.PI * 3f / 4f
+						Rotation = MathF.PI * 3f / 4f
 					}
 				},
 				{
@@ -5497,7 +_,7 @@
 					175,
 					new NPCBestiaryDrawModifiers(0) {
 						Position = new Vector2(4f, -2f),
-						Rotation = (float)Math.PI * 3f / 4f
+						Rotation = MathF.PI * 3f / 4f
 					}
 				},
 				{
