--- decompiled/TerrariaBuildable/Terraria/Audio/LegacyAudioSystem.cs
+++ decompiled/TerrariaUnified/Terraria/Audio/LegacyAudioSystem.cs
@@ -69,6 +_,12 @@
 	private IAudioTrack FindReplacementTrack(List<IContentSource> sources, string assetPath)
 	{
 		IAudioTrack audioTrack = null;
+		if (Main.dedServ) {
+			for (int i = 0; i < sources.Count; i++) {
+				sources[i].HasAsset(assetPath);
+			}
+			return null;
+		}
 		for (int i = 0; i < sources.Count; i++) {
 			IContentSource contentSource = sources[i];
 			if (!contentSource.HasAsset(assetPath)) {
