--- decompiled/TerrariaServerWindows/Terraria/Terraria.csproj
+++ decompiled/TerrariaServerLinux/Terraria/Terraria.csproj
@@ -10,11 +_,8 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="CsvHelper" />
+    <Reference Include="FNA" />
     <Reference Include="Ionic.Zip.CF" />
-    <Reference Include="Microsoft.Xna.Framework" />
-    <Reference Include="Microsoft.Xna.Framework.Game" />
-    <Reference Include="Microsoft.Xna.Framework.Graphics" />
-    <Reference Include="Microsoft.Xna.Framework.Xact" />
     <Reference Include="Newtonsoft.Json" />
     <ProjectReference Include="../RailSDK.Net/RailSDK.Net.csproj" />
     <EmbeddedResource Include="../RailSDK.Net/bin/$(Configuration)/$(TargetFramework)/RailSDK.Net.dll" />
