--- decompiled/TerrariaServerWindows/Terraria/Localization/LanguageManager.cs
+++ decompiled/TerrariaServerLinux/Terraria/Localization/LanguageManager.cs
@@ -66,8 +_,8 @@
 			string textValue = GetTextValue(key);
 			textValue.Replace(",", "").Replace(".", "").Replace("\"", "")
 				.Trim();
-			string[] array = textValue.Split(' ');
-			string[] array2 = textValue.Split(' ');
+			string[] array = textValue.Split(new char[1] { ' ' });
+			string[] array2 = textValue.Split(new char[1] { ' ' });
 			if (array.Length == array2.Length)
 			{
 				string[] array3 = array;
@@ -181,7 +_,8 @@
 	public void UseSources(List<IContentSource> sourcesFromLowestToHighest)
 	{
 		string name = ActiveCulture.Name;
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
-		string assetNameStart = ("Localization" + Path.DirectorySeparatorChar + name).ToLower();
+		string assetNameStart = ("Localization" + directorySeparatorChar + name).ToLower();
 		LoadLanguage(ActiveCulture, processCopyCommands: false);
 		foreach (IContentSource item in sourcesFromLowestToHighest)
 		{
