--- decompiled/TerrariaServerWindows/Terraria/Localization/LanguageManager.cs
+++ decompiled/TerrariaServerLinux/Terraria/Localization/LanguageManager.cs
@@ -160,7 +_,8 @@
 	public void UseSources(List<IContentSource> sourcesFromLowestToHighest)
 	{
 		string name = ActiveCulture.Name;
+		char directorySeparatorChar = Path.DirectorySeparatorChar;
-		string assetNameStart = ("Localization" + Path.DirectorySeparatorChar + name).ToLower();
+		string assetNameStart = ("Localization" + directorySeparatorChar + name).ToLower();
 		LoadLanguage(ActiveCulture, processCopyCommands: false);
 		foreach (IContentSource item in sourcesFromLowestToHighest) {
 			foreach (string item2 in item.GetAllAssetsStartingWith(assetNameStart)) {
