--- decompiled/TerrariaClientWindows/ReLogic/Peripherals/RGB/RgbDevice.cs
+++ decompiled/TerrariaClientLinux/ReLogic/Peripherals/RGB/RgbDevice.cs
@@ -106,11 +_,11 @@
 		for (int i = 0; i < LedCount; i++)
 		{
 			Vector2 canvasPositionOfIndex = _backBuffer.GetCanvasPositionOfIndex(i);
-			Vector4 vector = _backBuffer.Colors[i];
+			Vector4 color = _backBuffer.Colors[i];
-			vector.X *= vector.W;
+			color.X *= color.W;
-			vector.Y *= vector.W;
+			color.Y *= color.W;
-			vector.Z *= vector.W;
+			color.Z *= color.W;
-			drawer.DrawSquare(new Vector4(canvasPositionOfIndex * scale + position, scale / 10f, scale / 10f), new Color(vector));
+			drawer.DrawSquare(new Vector4(canvasPositionOfIndex * scale + position, scale / 10f, scale / 10f), new Color(color));
 		}
 	}
 }
