--- decompiled/TerrariaServerWindows/Terraria/UI/Chat/ChatManager.cs
+++ decompiled/TerrariaServerLinux/Terraria/UI/Chat/ChatManager.cs
@@ -127,10 +_,10 @@
 				result.Y = Math.Max(result.Y, vector.Y + size.Y);
 				continue;
 			}
-			string[] array = textSnippet.Text.Split('\n');
+			string[] array = textSnippet.Text.Split(new char[1] { '\n' });
 			string[] array2 = array;
 			for (int j = 0; j < array2.Length; j++) {
-				string[] array3 = array2[j].Split(' ');
+				string[] array3 = array2[j].Split(new char[1] { ' ' });
 				for (int k = 0; k < array3.Length; k++) {
 					if (k != 0) {
 						vector.X += x * baseScale.X * num;
@@ -196,12 +_,12 @@
 				result.X = Math.Max(result.X, vector.X);
 				continue;
 			}
-			string[] array = textSnippet.Text.Split('\n');
+			string[] array = textSnippet.Text.Split(new char[1] { '\n' });
 			array = Regex.Split(textSnippet.Text, "(\n)");
 			bool flag = true;
 			foreach (string text in array) {
 				string[] array2 = Regex.Split(text, "( )");
-				array2 = text.Split(' ');
+				array2 = text.Split(new char[1] { ' ' });
 				if (text == "\n") {
 					vector.Y += (float)font.LineSpacing * num3 * baseScale.Y;
 					vector.X = position.X;
@@ -270,14 +_,14 @@
 	{
 		Vector2 vector = position;
 		Vector2 result = vector;
-		string[] array = text.Split('\n');
+		string[] array = text.Split(new char[1] { '\n' });
 		float x = font.MeasureString(" ").X;
 		Color color = baseColor;
 		float num = 1f;
 		float num2 = 0f;
 		string[] array2 = array;
 		for (int i = 0; i < array2.Length; i++) {
-			string[] array3 = array2[i].Split(':');
+			string[] array3 = array2[i].Split(new char[1] { ':' });
 			foreach (string text2 in array3) {
 				if (text2.StartsWith("sss")) {
 					if (text2.StartsWith("sss1")) {
@@ -293,7 +_,7 @@
 					}
 					continue;
 				}
-				string[] array4 = text2.Split(' ');
+				string[] array4 = text2.Split(new char[1] { ' ' });
 				for (int k = 0; k < array4.Length; k++) {
 					if (k != 0) {
 						vector.X += x * baseScale.X * num;
