--- decompiled/TerrariaServerWindows/Terraria/GameContent/RGB/CultistShader.cs
+++ decompiled/TerrariaServerMac/Terraria/GameContent/RGB/CultistShader.cs
@@ -35,16 +_,16 @@
 			float num = time * 0.5f + canvasPositionOfIndex.X + canvasPositionOfIndex.Y;
 			float value = (float)Math.Cos(num) * 2f + 2f;
 			value = MathHelper.Clamp(value, 0f, 1f);
-			num = (num + (float)Math.PI) % ((float)Math.PI * 6f);
+			num = (num + MathF.PI) % (MathF.PI * 6f);
 			Vector4 value2;
-			if (num < (float)Math.PI * 2f)
+			if (num < MathF.PI * 2f)
 			{
 				float staticNoise = NoiseHelper.GetStaticNoise(canvasPositionOfIndex * 0.3f + new Vector2(12.5f, time * 0.2f));
 				staticNoise = Math.Max(0f, 1f - staticNoise * staticNoise * 4f * staticNoise);
 				staticNoise = MathHelper.Clamp(staticNoise, 0f, 1f);
 				value2 = Vector4.Lerp(_fireDarkColor, _fireBrightColor, staticNoise);
 			}
-			else if (num < (float)Math.PI * 4f)
+			else if (num < MathF.PI * 4f)
 			{
 				float dynamicNoise = NoiseHelper.GetDynamicNoise(new Vector2((canvasPositionOfIndex.X + canvasPositionOfIndex.Y) * 0.2f, 0f), time / 5f);
 				dynamicNoise = Math.Max(0f, 1f - dynamicNoise * 1.5f);
