--- decompiled/TerrariaServerWindows/Terraria/GameContent/RGB/TwinsShader.cs
+++ decompiled/TerrariaServerMac/Terraria/GameContent/RGB/TwinsShader.cs
@@ -82,7 +_,7 @@
 				}
 				Vector4 value = Vector4.Lerp(_eyeColor, _veinColor, num3);
 				float value2 = (float)Math.Atan2(vector3.Y, vector3.X);
-				if (!flag && (float)Math.PI - Math.Abs(value2) < 0.6f)
+				if (!flag && MathF.PI - Math.Abs(value2) < 0.6f)
 				{
 					value = _mouthColor;
 				}
