--- decompiled/TerrariaServerWindows/ReLogic/Peripherals/RGB/SteelSeries/GameSenseConnection.cs
+++ decompiled/TerrariaServerLinux/ReLogic/Peripherals/RGB/SteelSeries/GameSenseConnection.cs
@@ -320,7 +_,7 @@
 				JObject jObject = JObject.Parse(text);
 				coreProps coreProps = default(coreProps);
 				coreProps.address = (string)jObject.GetValue("address");
-				string[] array = coreProps.address.Split(':');
+				string[] array = coreProps.address.Split(new char[1] { ':' });
 				Convert.ToUInt16(array[1]);
 				result = array[1];
 			} catch (Exception e) {
