--- decompiled/TerrariaServerWindows/ReLogic/Localization/IME/FnaIme.cs
+++ decompiled/TerrariaServerMac/ReLogic/Localization/IME/FnaIme.cs
@@ -1,3 +_,5 @@
+using Microsoft.Xna.Framework.Input;
+
 namespace ReLogic.Localization.IME;
 
 internal class FnaIme : PlatformIme
@@ -12,6 +_,11 @@
 
 	public override uint SelectedCandidate => 0u;
 
+	public FnaIme()
+	{
+		TextInputEXT.TextInput += OnCharCallback;
+	}
+
 	private void OnCharCallback(char key)
 	{
 		if (base.IsEnabled) {
@@ -27,6 +_,9 @@
 	protected override void Dispose(bool disposing)
 	{
 		if (!_disposedValue) {
+			if (disposing) {
+				TextInputEXT.TextInput -= OnCharCallback;
+			}
 			if (base.IsEnabled) {
 				Disable();
 			}
