--- decompiled/TerrariaClientWindows/ReLogic/Utilities/MultiTimer.cs
+++ decompiled/TerrariaClientMac/ReLogic/Utilities/MultiTimer.cs
@@ -87,7 +_,28 @@
 			foreach (KeyValuePair<string, TimerData> item2 in _timerDataMap) {
 				TimerData value = item2.Value;
 				string text = new string(' ', num2 - item2.Key.Length);
-				Console.WriteLine(item2.Key + text + " : (Average: " + value.Average.ToString("F4") + " Min: " + value.Min.ToString("F4") + " Max: " + value.Max.ToString("F4") + " from " + (int)value.Ticks + " records)");
+				object[] obj = new object[11] {
+					item2.Key,
+					text,
+					" : (Average: ",
+					value.Average.ToString("F4"),
+					" Min: ",
+					null,
+					null,
+					null,
+					null,
+					null,
+					null
+				};
+				double min = value.Min;
+				obj[5] = min.ToString("F4");
+				obj[6] = " Max: ";
+				min = value.Max;
+				obj[7] = min.ToString("F4");
+				obj[8] = " from ";
+				obj[9] = (int)value.Ticks;
+				obj[10] = " records)";
+				Console.WriteLine(string.Concat(obj));
 				num += value.Total;
 			}
 			_timerDataMap.Clear();
