--- decompiled/TerrariaServerWindows/Terraria/Graphics/VertexStrip.cs
+++ decompiled/TerrariaServerMac/Terraria/Graphics/VertexStrip.cs
@@ -84,7 +_,7 @@
 			}
 			Vector2 vector2 = positions[i + 1];
 			float num3 = MathHelper.WrapAngle(rotations[i + 1]);
-			int num4 = (int)(Math.Abs(MathHelper.WrapAngle(num3 - num2)) / ((float)Math.PI / 12f));
+			int num4 = (int)(Math.Abs(MathHelper.WrapAngle(num3 - num2)) / (MathF.PI / 12f));
 			if (num4 != 0) {
 				float num5 = vector.Distance(vector2);
 				Vector2 value = vector + num2.ToRotationVector2() * num5;
@@ -149,7 +_,7 @@
 		}
 		Color color = colorFunction(progressOnStrip);
 		float num = widthFunction(progressOnStrip);
-		Vector2 vector = MathHelper.WrapAngle(rot - (float)Math.PI / 2f).ToRotationVector2() * num;
+		Vector2 vector = MathHelper.WrapAngle(rot - MathF.PI / 2f).ToRotationVector2() * num;
 		_vertices[indexOnVertexArray].Position = pos + vector;
 		_vertices[indexOnVertexArray + 1].Position = pos - vector;
 		_vertices[indexOnVertexArray].TexCoord = new Vector2(progressOnStrip, 1f);
