--- decompiled/TerrariaClientWindows/Terraria/GameContent/Events/MoonlordDeathDrama.cs
+++ decompiled/TerrariaClientLinux/Terraria/GameContent/Events/MoonlordDeathDrama.cs
@@ -240,13 +_,13 @@
 	public static void ThrowPieces(Vector2 MoonlordCoreCenter, int DramaSeed)
 	{
 		UnifiedRandom r = new UnifiedRandom(DramaSeed);
-		Vector2 vector = Vector2.UnitY.RotatedBy(r.NextFloat() * ((float)Math.PI / 2f) - (float)Math.PI / 4f + (float)Math.PI);
+		Vector2 vector = Vector2.UnitY.RotatedBy(r.NextFloat() * (MathF.PI / 2f) - MathF.PI / 4f + MathF.PI);
 		_pieces.Add(new MoonlordPiece(Main.Assets.Request<Texture2D>("Images/Misc/MoonExplosion/Spine").Value, new Vector2(64f, 150f), MoonlordCoreCenter + new Vector2(0f, 50f), vector * 6f, 0f, r.NextFloat() * 0.1f - 0.05f));
-		vector = Vector2.UnitY.RotatedBy(r.NextFloat() * ((float)Math.PI / 2f) - (float)Math.PI / 4f + (float)Math.PI);
+		vector = Vector2.UnitY.RotatedBy(r.NextFloat() * (MathF.PI / 2f) - MathF.PI / 4f + MathF.PI);
 		_pieces.Add(new MoonlordPiece(Main.Assets.Request<Texture2D>("Images/Misc/MoonExplosion/Shoulder").Value, new Vector2(40f, 120f), MoonlordCoreCenter + new Vector2(50f, -120f), vector * 10f, 0f, r.NextFloat() * 0.1f - 0.05f));
-		vector = Vector2.UnitY.RotatedBy(r.NextFloat() * ((float)Math.PI / 2f) - (float)Math.PI / 4f + (float)Math.PI);
+		vector = Vector2.UnitY.RotatedBy(r.NextFloat() * (MathF.PI / 2f) - MathF.PI / 4f + MathF.PI);
 		_pieces.Add(new MoonlordPiece(Main.Assets.Request<Texture2D>("Images/Misc/MoonExplosion/Torso").Value, new Vector2(192f, 252f), MoonlordCoreCenter, vector * 8f, 0f, r.NextFloat() * 0.1f - 0.05f));
-		vector = Vector2.UnitY.RotatedBy(r.NextFloat() * ((float)Math.PI / 2f) - (float)Math.PI / 4f + (float)Math.PI);
+		vector = Vector2.UnitY.RotatedBy(r.NextFloat() * (MathF.PI / 2f) - MathF.PI / 4f + MathF.PI);
 		_pieces.Add(new MoonlordPiece(Main.Assets.Request<Texture2D>("Images/Misc/MoonExplosion/Head").Value, new Vector2(138f, 185f), MoonlordCoreCenter - new Vector2(0f, 200f), vector * 12f, 0f, r.NextFloat() * 0.1f - 0.05f));
 	}
 
