--- decompiled/TerrariaClientWindows/Terraria/Net/SteamAddress.cs
+++ decompiled/TerrariaClientMac/Terraria/Net/SteamAddress.cs
@@ -30,7 +_,9 @@
 	{
 		if (Program.LaunchParameters.ContainsKey("-localsteamid"))
 		{
-			return Program.LaunchParameters["-localsteamid"].Equals(SteamId.m_SteamID.ToString());
+			string text = Program.LaunchParameters["-localsteamid"];
+			CSteamID steamId = SteamId;
+			return text.Equals(steamId.m_SteamID.ToString());
 		}
 		return false;
 	}
