--- decompiled/TerrariaClientWindows/Terraria/GameContent/PlayerSleepingHelper.cs
+++ decompiled/TerrariaClientLinux/Terraria/GameContent/PlayerSleepingHelper.cs
@@ -58,7 +_,7 @@
 		if (isSleeping != state) {
 			isSleeping = state;
 			if (state) {
-				player.fullRotation = (float)Math.PI / 2f * (float)(-player.direction);
+				player.fullRotation = MathF.PI / 2f * (float)(-player.direction);
 				return;
 			}
 			player.fullRotation = 0f;
