--- decompiled/TerrariaClientWindows/Terraria/Terraria.csproj
+++ decompiled/TerrariaBuildable/Terraria/Terraria.csproj
@@ -1,44 +_,44 @@
 <Project Sdk="Microsoft.NET.Sdk">
-  <Import Project="../Configuration.targets" />
-  <PropertyGroup>
-    <OutputType>WinExe</OutputType>
-    <Nullable>enable</Nullable>
-    <Version>1.4.4.9</Version>
-    <Company>Re-Logic</Company>
-    <Company>Copyright © 2022 Re-Logic</Company>
-    <RootNamespace>Terraria</RootNamespace>
-  </PropertyGroup>
-  <ItemGroup>
-    <Reference Include="CsvHelper" />
-    <Reference Include="Ionic.Zip.CF" />
-    <Reference Include="Microsoft.Xna.Framework" />
-    <Reference Include="Microsoft.Xna.Framework.Game" />
-    <Reference Include="Microsoft.Xna.Framework.Graphics" />
-    <Reference Include="Microsoft.Xna.Framework.Xact" />
-    <Reference Include="MP3Sharp" />
-    <Reference Include="Newtonsoft.Json" />
-    <Reference Include="NVorbis" />
-    <ProjectReference Include="../RailSDK.Net/RailSDK.Net.csproj" />
-    <EmbeddedResource Include="../RailSDK.Net/bin/$(Configuration)/$(TargetFramework)/RailSDK.Net.dll" />
-    <ProjectReference Include="../ReLogic/ReLogic.csproj" />
-    <EmbeddedResource Include="../ReLogic/bin/$(Configuration)/$(TargetFramework)/ReLogic.dll" />
-    <Reference Include="Steamworks.NET" />
-    <ProjectReference Include="../SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj" />
-    <EmbeddedResource Include="../SteelSeriesEngineWrapper/bin/$(Configuration)/$(TargetFramework)/SteelSeriesEngineWrapper.dll" />
-    <Reference Include="System" />
-    <Reference Include="System.Core" />
-    <Reference Include="System.Drawing" />
-    <Reference Include="System.Runtime.Serialization" />
-    <Reference Include="System.Windows.Forms" />
-    <Reference Include="WindowsBase" />
-  </ItemGroup>
-  <ItemGroup>
-    <EmbeddedResource Include="GameContent/Creative/Content/*" />
-    <EmbeddedResource Include="GameContent/Metadata/MaterialData/*" />
-    <EmbeddedResource Include="GameContent/WorldBuilding/*" />
-    <EmbeddedResource Include="IO/Data/*" />
-    <EmbeddedResource Include="Libraries/**" />
-    <EmbeddedResource Include="Localization/Content/**" />
-    <EmbeddedResource Include="Microsoft/**" />
-  </ItemGroup>
+
+    <Import Project="../Configuration.targets"/>
+
+    <PropertyGroup>
+        <OutputType>WinExe</OutputType>
+        <Version>1.4.4.9</Version>
+        <Company>Re-Logic</Company>
+        <Company>Copyright © 2022 Re-Logic</Company>
+        <RootNamespace>Terraria</RootNamespace>
+        <TargetFramework>net40</TargetFramework>
+        <LangVersion>latest</LangVersion>
+        <PlatformTarget>x86</PlatformTarget>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <ProjectReference Include="../RailSDK/RailSDK.csproj"/>
+        <ProjectReference Include="../ReLogic/ReLogic.csproj"/>
+        <ProjectReference Include="../SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj"/>
+        <Reference Include="CsvHelper"/>
+        <Reference Include="Ionic.Zip.CF"/>
+        <Reference Include="Microsoft.Xna.Framework"/>
+        <Reference Include="Microsoft.Xna.Framework.Game"/>
+        <Reference Include="Microsoft.Xna.Framework.Graphics"/>
+        <Reference Include="Microsoft.Xna.Framework.Xact"/>
+        <Reference Include="MP3Sharp"/>
+        <Reference Include="Newtonsoft.Json"/>
+        <Reference Include="NVorbis"/>
+        <Reference Include="NVorbis.System.ValueTuple"/>
+        <Reference Include="Steamworks.NET"/>
+        <Reference Include="System.Windows.Forms"/>
+        <Reference Include="WindowsBase"/>
+    </ItemGroup>
+
+    <ItemGroup>
+        <EmbeddedResource Include="GameContent/Creative/Content/*"/>
+        <EmbeddedResource Include="GameContent/Metadata/MaterialData/*"/>
+        <EmbeddedResource Include="GameContent/WorldBuilding/*"/>
+        <EmbeddedResource Include="IO/Data/*"/>
+        <EmbeddedResource Include="Localization/Content/**"/>
+        <EmbeddedResource Include="Microsoft/**"/>
+    </ItemGroup>
+
 </Project>
