--- decompiled/TerrariaClientWindows/SteelSeriesEngineWrapper/FullSerializer/Internal/fsReflectedConverter.cs
+++ decompiled/TerrariaBuildable/SteelSeriesEngineWrapper/FullSerializer/Internal/fsReflectedConverter.cs
@@ -35,7 +_,7 @@
 
 	public override fsResult TryDeserialize(fsData data, ref object instance, Type storageType)
 	{
-		fsResult success = fsResult.Success;
+		fsResult success = FullSerializer.fsResult.Success;
 		fsResult fsResult = (success += CheckType(data, fsDataType.Object));
 		if (fsResult.Failed) {
 			return success;
