--- decompiled/TerrariaBuildable/ReLogic/ReLogic.csproj
+++ decompiled/TerrariaUnified/ReLogic/ReLogic.csproj
@@ -1,28 +_,25 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
-    <Import Project="../Configuration.targets"/>
+    <Import Project="../Configuration.targets" />
 
     <PropertyGroup>
         <Company>Re-Logic</Company>
         <Company>Copyright © Re-Logic 2017</Company>
         <RootNamespace>ReLogic</RootNamespace>
-        <TargetFramework>net40</TargetFramework>
+        <TargetFramework>net6.0</TargetFramework>
         <LangVersion>latest</LangVersion>
     </PropertyGroup>
 
     <ItemGroup>
-        <ProjectReference Include="../RailSDK/RailSDK.csproj"/>
+        <ProjectReference Include="../RailSDK/RailSDK.csproj" />
-        <ProjectReference Include="../SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj"/>
+        <ProjectReference Include="../SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj" />
+        <ProjectReference Include="../../../src\FNA\FNA.csproj" />
-        <Reference Include="../Terraria/Libraries/DotNetZip/Ionic.Zip.CF.dll"/>
+        <Reference Include="../Terraria/Libraries/DotNetZip/Ionic.Zip.CF.dll" />
-        <Reference Include="../Terraria/Libraries/XNA/Microsoft.Xna.Framework.dll"/>
+        <Reference Include="../Terraria/Libraries/JSON.NET/Newtonsoft.Json.dll" />
-        <Reference Include="../Terraria/Libraries/XNA/Microsoft.Xna.Framework.Game.dll"/>
-        <Reference Include="../Terraria/Libraries/XNA/Microsoft.Xna.Framework.Graphics.dll"/>
-        <Reference Include="../Terraria/Libraries/JSON.NET/Newtonsoft.Json.dll"/>
-        <Reference Include="System.Windows.Forms"/>
     </ItemGroup>
 
     <ItemGroup>
-        <EmbeddedResource Include="Microsoft/**"/>
+        <EmbeddedResource Include="Microsoft/**" />
     </ItemGroup>
 
 </Project>
