--- decompiled/TerrariaClientWindows/Terraria/DelegateMethods.cs
+++ decompiled/TerrariaClientMac/Terraria/DelegateMethods.cs
@@ -17,7 +_,7 @@
 			Float(proj, walking);
 			if (walking) {
 				float num = (float)Main.timeForVisualEffects % 90f / 90f;
-				proj.localAI[1] = (float)Math.PI * 2f * num;
+				proj.localAI[1] = MathF.PI * 2f * num;
 			} else {
 				proj.localAI[1] = 0f;
 			}
@@ -31,7 +_,7 @@
 				float num = Utils.MultiLerp(percent, 0f, 0f, 16f, 20f, 20f, 16f, 0f, 0f);
 				float num2 = Utils.MultiLerp(percent2, 0f, 0f, 0.25f, 0.25f, 0.5f, 0.5f, 0.75f, 0.75f, 1f, 1f);
 				proj.position.Y -= num;
-				proj.rotation = (float)Math.PI * 2f * num2;
+				proj.rotation = MathF.PI * 2f * num2;
 			} else {
 				proj.rotation = 0f;
 			}
@@ -70,12 +_,12 @@
 				}
 				vector = vector.RotatedBy(-0.05235987901687622);
 			}
-			proj.rotation = vector.ToRotation() + (float)Math.PI / 10f + (float)Math.PI;
+			proj.rotation = vector.ToRotation() + MathF.PI / 10f + MathF.PI;
 			if (proj.type == 887) {
-				proj.rotation += (float)Math.PI / 8f;
+				proj.rotation += MathF.PI / 8f;
 			}
 			if (proj.type == 893) {
-				proj.rotation += (float)Math.PI / 2f;
+				proj.rotation += MathF.PI / 2f;
 			}
 		}
 
@@ -83,7 +_,7 @@
 		{
 			Float(proj, walking);
 			if (walking) {
-				proj.rotation = (float)Math.PI * 2f * ((float)Main.timeForVisualEffects % 20f / 20f);
+				proj.rotation = MathF.PI * 2f * ((float)Main.timeForVisualEffects % 20f / 20f);
 			} else {
 				proj.rotation = 0f;
 			}
@@ -99,13 +_,13 @@
 		{
 			float num = 0.5f;
 			float num2 = (float)Main.timeForVisualEffects % 60f / 60f;
-			proj.position.Y += 0f - num + (float)(Math.Cos(num2 * ((float)Math.PI * 2f) * 2f) * (double)(num * 2f));
+			proj.position.Y += 0f - num + (float)(Math.Cos(num2 * (MathF.PI * 2f) * 2f) * (double)(num * 2f));
 		}
 
 		public static void RotateForwardWhenWalking(Projectile proj, bool walking)
 		{
 			if (walking) {
-				proj.rotation = (float)Math.PI / 6f;
+				proj.rotation = MathF.PI / 6f;
 			} else {
 				proj.rotation = 0f;
 			}
