--- decompiled/TerrariaClientWindows/Terraria/Social/Steam/WorkshopHelper.cs
+++ decompiled/TerrariaClientMac/Terraria/Social/Steam/WorkshopHelper.cs
@@ -80,7 +_,8 @@
 					}
 					try
 					{
+						char directorySeparatorChar = Path.DirectorySeparatorChar;
-						string path = listOfSubscribedItemsPath + Path.DirectorySeparatorChar + "workshop.json";
+						string path = listOfSubscribedItemsPath + directorySeparatorChar + "workshop.json";
 						if (!File.Exists(path))
 						{
 							continue;
@@ -213,7 +_,8 @@
 					Visibility = visibility
 				};
 				ulong? num = null;
+				char directorySeparatorChar = Path.DirectorySeparatorChar;
-				if (AWorkshopEntry.TryReadingManifest(contentFolderPath + Path.DirectorySeparatorChar + "workshop.json", out var info))
+				if (AWorkshopEntry.TryReadingManifest(contentFolderPath + directorySeparatorChar + "workshop.json", out var info))
 				{
 					num = info.workshopEntryId;
 				}
@@ -348,7 +_,8 @@
 
 			private bool TryWritingManifestToFolder(string folderPath, string manifestText)
 			{
+				char directorySeparatorChar = Path.DirectorySeparatorChar;
-				string path = folderPath + Path.DirectorySeparatorChar + "workshop.json";
+				string path = folderPath + directorySeparatorChar + "workshop.json";
 				bool result = true;
 				try
 				{
@@ -416,11 +_,17 @@
 			{
 				if (_world.IsCloudSave)
 				{
-					FileUtilities.CopyToLocal(_world.Path, _entryData.ContentFolderPath + Path.DirectorySeparatorChar + "world.wld");
+					string path = _world.Path;
+					string contentFolderPath = _entryData.ContentFolderPath;
+					char directorySeparatorChar = Path.DirectorySeparatorChar;
+					FileUtilities.CopyToLocal(path, contentFolderPath + directorySeparatorChar + "world.wld");
 				}
 				else
 				{
-					FileUtilities.Copy(_world.Path, _entryData.ContentFolderPath + Path.DirectorySeparatorChar + "world.wld", cloud: false);
+					string path2 = _world.Path;
+					string contentFolderPath2 = _entryData.ContentFolderPath;
+					char directorySeparatorChar = Path.DirectorySeparatorChar;
+					FileUtilities.Copy(path2, contentFolderPath2 + directorySeparatorChar + "world.wld", cloud: false);
 				}
 			}
 		}
