--- decompiled/TerrariaBuildable/Terraria/Terraria.csproj
+++ decompiled/TerrariaUnified/Terraria/Terraria.csproj
@@ -1,6 +_,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
-    <Import Project="../Configuration.targets"/>
+    <Import Project="../Configuration.targets" />
 
     <PropertyGroup>
         <OutputType>WinExe</OutputType>
@@ -8,37 +_,30 @@
         <Company>Re-Logic</Company>
         <Company>Copyright © 2022 Re-Logic</Company>
         <RootNamespace>Terraria</RootNamespace>
-        <TargetFramework>net40</TargetFramework>
+        <TargetFramework>net6.0</TargetFramework>
         <LangVersion>latest</LangVersion>
-        <PlatformTarget>x86</PlatformTarget>
     </PropertyGroup>
 
     <ItemGroup>
-        <ProjectReference Include="../RailSDK/RailSDK.csproj"/>
+        <ProjectReference Include="../RailSDK/RailSDK.csproj" />
-        <ProjectReference Include="../ReLogic/ReLogic.csproj"/>
+        <ProjectReference Include="../ReLogic/ReLogic.csproj" />
-        <ProjectReference Include="../SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj"/>
+        <ProjectReference Include="../SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj" />
-        <Reference Include="CsvHelper"/>
+        <Reference Include="CsvHelper" />
-        <Reference Include="Ionic.Zip.CF"/>
+        <Reference Include="Ionic.Zip.CF" />
-        <Reference Include="Microsoft.Xna.Framework"/>
-        <Reference Include="Microsoft.Xna.Framework.Game"/>
-        <Reference Include="Microsoft.Xna.Framework.Graphics"/>
-        <Reference Include="Microsoft.Xna.Framework.Xact"/>
-        <Reference Include="MP3Sharp"/>
+        <Reference Include="MP3Sharp" />
-        <Reference Include="Newtonsoft.Json"/>
+        <Reference Include="Newtonsoft.Json" />
-        <Reference Include="NVorbis"/>
+        <Reference Include="NVorbis" />
-        <Reference Include="NVorbis.System.ValueTuple"/>
+        <Reference Include="NVorbis.System.ValueTuple" />
-        <Reference Include="Steamworks.NET"/>
+        <Reference Include="Steamworks.NET" />
-        <Reference Include="System.Windows.Forms"/>
-        <Reference Include="WindowsBase"/>
     </ItemGroup>
 
     <ItemGroup>
-        <EmbeddedResource Include="GameContent/Creative/Content/*"/>
+        <EmbeddedResource Include="GameContent/Creative/Content/*" />
-        <EmbeddedResource Include="GameContent/Metadata/MaterialData/*"/>
+        <EmbeddedResource Include="GameContent/Metadata/MaterialData/*" />
-        <EmbeddedResource Include="GameContent/WorldBuilding/*"/>
+        <EmbeddedResource Include="GameContent/WorldBuilding/*" />
-        <EmbeddedResource Include="IO/Data/*"/>
+        <EmbeddedResource Include="IO/Data/*" />
-        <EmbeddedResource Include="Localization/Content/**"/>
+        <EmbeddedResource Include="Localization/Content/**" />
-        <EmbeddedResource Include="Microsoft/**"/>
+        <EmbeddedResource Include="Microsoft/**" />
     </ItemGroup>
 
 </Project>
