--- decompiled/TerrariaBuildable/Terraria/WindowsLaunch.cs
+++ decompiled/TerrariaUnified/Terraria/WindowsLaunch.cs
@@ -1,5 +_,6 @@
 using System;
 using System.IO;
+using System.Linq;
 using System.Reflection;
 using System.Runtime.InteropServices;
 using Terraria.Social;
@@ -62,6 +_,12 @@
 			stream.Read(array, 0, array.Length);
 			return Assembly.Load(array);
 		};
-		Program.LaunchGame(args);
+		Environment.SetEnvironmentVariable("FNA_WORKAROUND_WINDOW_RESIZABLE", "1");
+		bool dedServ = args.Contains("-server");
+		if (dedServ) {
+			_handleRoutine = ConsoleCtrlCheck;
+			SetConsoleCtrlHandler(_handleRoutine, add: true);
+		}
+		Program.LaunchGame(args, monoArgs: !OperatingSystem.IsWindows());
 	}
 }
