--- decompiled/TerrariaServerWindows/Terraria/GameContent/Creative/CreativePowerManager.cs
+++ decompiled/TerrariaServerLinux/Terraria/GameContent/Creative/CreativePowerManager.cs
@@ -68,7 +_,7 @@
 		{
 			return;
 		}
-		string[] array = line.Substring(length).Split('=');
+		string[] array = line.Substring(length).Split(new char[1] { '=' });
 		if (array.Length == 2 && int.TryParse(array[1].Trim(), out var result))
 		{
 			PowerPermissionLevel powerPermissionLevel = (PowerPermissionLevel)Utils.Clamp(result, 0, 2);
