--- decompiled/TerrariaServerWindows/Terraria/GameContent/Creative/CreativeItemSacrificesCatalog.cs
+++ decompiled/TerrariaServerLinux/Terraria/GameContent/Creative/CreativeItemSacrificesCatalog.cs
@@ -21,7 +_,7 @@
 			if (text.StartsWith("//")) {
 				continue;
 			}
-			string[] array2 = text.Split('\t');
+			string[] array2 = text.Split(new char[1] { '\t' });
 			if (array2.Length >= 3 && ItemID.Search.TryGetId(array2[0], out var id)) {
 				int value = 0;
 				bool flag = false;
