--- decompiled/TerrariaClientWindows/Terraria/Main.cs
+++ decompiled/TerrariaServerWindows/Terraria/Main.cs
@@ -66,6 +_,7 @@
 using Terraria.Map;
 using Terraria.Net;
 using Terraria.ObjectData;
+using Terraria.Server;
 using Terraria.Social;
 using Terraria.UI;
 using Terraria.UI.Chat;
@@ -75,7 +_,7 @@
 
 namespace Terraria;
 
-public class Main : Game
+public class Main : Terraria.Server.Game
 {
 	public delegate void OnPlayerSelected(PlayerFileData player);
 
@@ -5224,7 +_,7 @@
 		if (!dedServ) {
 			Map = new WorldMap(maxTilesX, maxTilesY);
 			Configuration.Load();
-			graphics = new GraphicsDeviceManager(this as Game);
+			graphics = new GraphicsDeviceManager(this as Microsoft.Xna.Framework.Game);
 			base.Content.RootDirectory = "Content";
 		}
 	}
@@ -9501,8 +_,8 @@
 			if (dedServ) {
 				return;
 			}
-			bool flag = base.IsActive;
+			bool isActive = base.IsActive;
-			if (!flag) {
+			if (!isActive) {
 				audioSystem.PauseAll();
 				SoundEngine.StopAmbientSounds();
 			} else {
@@ -9847,7 +_,7 @@
 						if (remixWorld) {
 							if (cloudAlpha > 0f && (double)(player[myPlayer].position.Y / 16f) > rockLayer && player[myPlayer].position.Y / 16f < (float)(maxTilesY - 350) && !player[myPlayer].ZoneSnow && !player[myPlayer].ZoneDungeon) {
 								float trackVolume5 = musicFade[i];
-								audioSystem.UpdateAmbientCueState(i, flag, ref trackVolume5, ambientVolume * num8);
+								audioSystem.UpdateAmbientCueState(i, isActive, ref trackVolume5, ambientVolume * num8);
 								musicFade[i] = trackVolume5;
 							} else {
 								float trackVolume6 = musicFade[i];
@@ -9856,7 +_,7 @@
 							}
 						} else if (cloudAlpha > 0f && (double)player[myPlayer].position.Y < worldSurface * 16.0 + (double)(screenHeight / 2) && !player[myPlayer].ZoneSnow) {
 							float trackVolume7 = musicFade[i];
-							audioSystem.UpdateAmbientCueState(i, flag, ref trackVolume7, ambientVolume * num8);
+							audioSystem.UpdateAmbientCueState(i, isActive, ref trackVolume7, ambientVolume * num8);
 							musicFade[i] = trackVolume7;
 						} else {
 							float trackVolume8 = musicFade[i];
@@ -9880,7 +_,7 @@
 						if (remixWorld) {
 							if (!gameMenu && Math.Abs(windSpeedCurrent) >= num6 / 50f && (double)(player[myPlayer].position.Y / 16f) > rockLayer && player[myPlayer].position.Y / 16f < (float)(maxTilesY - 350) && !player[myPlayer].ZoneDungeon) {
 								float trackVolume = musicFade[i];
-								audioSystem.UpdateAmbientCueState(i, flag, ref trackVolume, ambientVolume * num5);
+								audioSystem.UpdateAmbientCueState(i, isActive, ref trackVolume, ambientVolume * num5);
 								musicFade[i] = trackVolume;
 							} else {
 								float trackVolume2 = musicFade[i];
@@ -9889,7 +_,7 @@
 							}
 						} else if (!gameMenu && Math.Abs(windSpeedCurrent) >= num6 / 50f && (double)player[myPlayer].position.Y < worldSurface * 16.0 + (double)(screenHeight / 2)) {
 							float trackVolume3 = musicFade[i];
-							audioSystem.UpdateAmbientCueState(i, flag, ref trackVolume3, ambientVolume * num5);
+							audioSystem.UpdateAmbientCueState(i, isActive, ref trackVolume3, ambientVolume * num5);
 							musicFade[i] = trackVolume3;
 						} else {
 							float trackVolume4 = musicFade[i];
@@ -9901,7 +_,7 @@
 					default: {
 						float tempFade = musicFade[i];
 						if (i == curMusic) {
-							audioSystem.UpdateCommonTrack(flag, i, num2, ref tempFade);
+							audioSystem.UpdateCommonTrack(isActive, i, num2, ref tempFade);
 						} else {
 							audioSystem.UpdateCommonTrackTowardStopping(i, num2, ref tempFade, isMainTrackAudible);
 						}
@@ -13650,7 +_,7 @@
 		if (netMode == 2) {
 			cloudAlpha = maxRaining;
 		}
-		bool flag2 = base.IsActive;
+		bool isActive = base.IsActive;
 		if (netMode == 1) {
 			TrySyncingMyPlayer();
 		}
@@ -13670,7 +_,7 @@
 		if (LocalGolfState != null) {
 			LocalGolfState.Update();
 		}
-		if ((flag2 || netMode == 1) && cloudAlpha > 0f) {
+		if ((isActive || netMode == 1) && cloudAlpha > 0f) {
 			Rain.MakeRain();
 		}
 		if (netMode != 1) {
@@ -46171,7 +_,7 @@
 	public void UpdateWeather(GameTime gameTime, int currentDayRateIteration)
 	{
 		if (netMode != 2 && currentDayRateIteration == 0) {
-			bool flag = base.IsActive;
+			bool isActive = base.IsActive;
 			if (thunderDelay > 0) {
 				thunderDelay--;
 				if (thunderDelay == 0) {
@@ -46185,7 +_,7 @@
 					if (center.Y > (float)num2) {
 						center.Y = num2;
 					}
-					if (flag) {
+					if (isActive) {
 						SoundEngine.PlaySound(43, center);
 					}
 				}
@@ -49755,7 +_,7 @@
 
 	private void DrawRainInMenu()
 	{
-		bool flag = base.IsActive;
+		bool isActive = base.IsActive;
 		Microsoft.Xna.Framework.Rectangle[] array = new Microsoft.Xna.Framework.Rectangle[6];
 		for (int i = 0; i < array.Length; i++) {
 			array[i] = new Microsoft.Xna.Framework.Rectangle(i * 4, 0, 2, 40);
@@ -49765,7 +_,7 @@
 			if (Main.rain[j].active) {
 				Rain rain = Main.rain[j];
 				spriteBatch.Draw(TextureAssets.Rain.Value, rain.position - screenPosition, array[rain.type], color, rain.rotation, Vector2.Zero, rain.scale, SpriteEffects.None, 0f);
-				if (flag) {
+				if (isActive) {
 					rain.Update();
 				}
 			}
