--- decompiled/TerrariaClientWindows/Terraria/Program.cs
+++ decompiled/TerrariaServerWindows/Terraria/Program.cs
@@ -1,6 +_,5 @@
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO;
 using System.Reflection;
 using System.Runtime.CompilerServices;
@@ -137,9 +_,6 @@
 
 	private static void InitializeConsoleOutput()
 	{
-		if (Debugger.IsAttached) {
-			return;
-		}
 		try {
 			Console.OutputEncoding = Encoding.UTF8;
 			if (Platform.IsWindows) {
@@ -171,6 +_,7 @@
 
 	public static void RunGame()
 	{
+		Main.dedServ = true;
 		LanguageManager.Instance.SetLanguage(GameCulture.DefaultCulture);
 		if (Platform.IsOSX) {
 			Main.OnEngineLoad += delegate {
