--- decompiled/TerrariaClientWindows/Terraria/Terraria.csproj
+++ decompiled/TerrariaServerWindows/Terraria/Terraria.csproj
@@ -6,7 +_,7 @@
     <Version>1.4.4.9</Version>
     <Company>Re-Logic</Company>
     <Company>Copyright © 2022 Re-Logic</Company>
-    <RootNamespace>Terraria</RootNamespace>
+    <RootNamespace>TerrariaServer</RootNamespace>
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="CsvHelper" />
@@ -15,16 +_,12 @@
     <Reference Include="Microsoft.Xna.Framework.Game" />
     <Reference Include="Microsoft.Xna.Framework.Graphics" />
     <Reference Include="Microsoft.Xna.Framework.Xact" />
-    <Reference Include="MP3Sharp" />
     <Reference Include="Newtonsoft.Json" />
-    <Reference Include="NVorbis" />
     <ProjectReference Include="../RailSDK.Net/RailSDK.Net.csproj" />
     <EmbeddedResource Include="../RailSDK.Net/bin/$(Configuration)/$(TargetFramework)/RailSDK.Net.dll" />
     <ProjectReference Include="../ReLogic/ReLogic.csproj" />
     <EmbeddedResource Include="../ReLogic/bin/$(Configuration)/$(TargetFramework)/ReLogic.dll" />
     <Reference Include="Steamworks.NET" />
-    <ProjectReference Include="../SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj" />
-    <EmbeddedResource Include="../SteelSeriesEngineWrapper/bin/$(Configuration)/$(TargetFramework)/SteelSeriesEngineWrapper.dll" />
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="System.Drawing" />
@@ -39,6 +_,5 @@
     <EmbeddedResource Include="IO/Data/*" />
     <EmbeddedResource Include="Libraries/**" />
     <EmbeddedResource Include="Localization/Content/**" />
-    <EmbeddedResource Include="Microsoft/**" />
   </ItemGroup>
 </Project>
