--- decompiled/TerrariaClientWindows/Terraria/GameContent/RGB/DeathShader.cs
+++ decompiled/TerrariaClientMac/Terraria/GameContent/RGB/DeathShader.cs
@@ -24,8 +_,8 @@
 	{
 		time *= 3f;
 		float amount = 0f;
-		float num = time % ((float)Math.PI * 4f);
+		float num = time % (MathF.PI * 4f);
-		if (num < (float)Math.PI) {
+		if (num < MathF.PI) {
 			amount = (float)Math.Sin(num);
 		}
 		for (int i = 0; i < fragment.Count; i++) {
