--- decompiled/TerrariaClientWindows/Terraria/GameContent/RGB/EmpressShader.cs
+++ decompiled/TerrariaClientMac/Terraria/GameContent/RGB/EmpressShader.cs
@@ -29,7 +_,7 @@
 			Point gridPositionOfIndex = fragment.GetGridPositionOfIndex(i);
 			Vector2 canvasPositionOfIndex = fragment.GetCanvasPositionOfIndex(i);
 			float staticNoise = NoiseHelper.GetStaticNoise(gridPositionOfIndex.X);
-			float num2 = MathHelper.Max(0f, (float)Math.Cos((staticNoise + num) * ((float)Math.PI * 2f) * 0.2f));
+			float num2 = MathHelper.Max(0f, (float)Math.Cos((staticNoise + num) * (MathF.PI * 2f) * 0.2f));
 			Vector4 value = Color.Lerp(Color.Black, Color.Indigo, 0.5f).ToVector4();
 			float num3 = Math.Max(0f, (float)Math.Sin(Main.GlobalTimeWrappedHourly * 2f + canvasPositionOfIndex.X * 1f));
 			num3 = 0f;
