--- decompiled/TerrariaClientWindows/Terraria/GameContent/UI/GameTipsDisplay.cs
+++ decompiled/TerrariaClientLinux/Terraria/GameContent/UI/GameTipsDisplay.cs
@@ -145,7 +_,7 @@
 			}
 			DynamicSpriteFont value = FontAssets.MouseText.Value;
 			string text = value.CreateWrappedText(currentTip.Text, num2, Language.ActiveCulture.CultureInfo);
-			if (text.Split('\n').Length > 2)
+			if (text.Split(new char[1] { '\n' }).Length > 2)
 			{
 				text = value.CreateWrappedText(currentTip.Text, num2 * 1.5f - 50f, Language.ActiveCulture.CultureInfo);
 			}
