--- decompiled/TerrariaClientWindows/Terraria/GameContent/Skies/LanternSky.cs
+++ decompiled/TerrariaClientMac/Terraria/GameContent/Skies/LanternSky.cs
@@ -242,12 +_,12 @@
 
 	private void DrawLantern(SpriteBatch spriteBatch, Lantern lantern, Color opacity, Vector2 depthScale, Vector2 position, float alpha)
 	{
-		float y = (Main.GlobalTimeWrappedHourly % 6f / 6f * ((float)Math.PI * 2f)).ToRotationVector2().Y;
+		float y = (Main.GlobalTimeWrappedHourly % 6f / 6f * (MathF.PI * 2f)).ToRotationVector2().Y;
 		float num = y * 0.2f + 0.8f;
 		Color color = new Color(255, 255, 255, 0) * _opacity * alpha * num * 0.4f;
 		for (float num2 = 0f; num2 < 1f; num2 += 1f / 3f)
 		{
-			Vector2 vector = new Vector2(0f, 2f).RotatedBy((float)Math.PI * 2f * num2 + lantern.Rotation) * y;
+			Vector2 vector = new Vector2(0f, 2f).RotatedBy(MathF.PI * 2f * num2 + lantern.Rotation) * y;
 			spriteBatch.Draw(lantern.Texture, position + vector, lantern.GetSourceRectangle(), color, lantern.Rotation, lantern.GetSourceRectangle().Size() / 2f, depthScale.X * 2f, SpriteEffects.None, 0f);
 		}
 		spriteBatch.Draw(lantern.Texture, position, lantern.GetSourceRectangle(), opacity * _opacity, lantern.Rotation, lantern.GetSourceRectangle().Size() / 2f, depthScale.X * 2f, SpriteEffects.None, 0f);
