--- decompiled/TerrariaClientWindows/Terraria/Graphics/WindowStateController.cs
+++ decompiled/TerrariaClientMac/Terraria/Graphics/WindowStateController.cs
@@ -20,7 +_,7 @@
 	public void TryMovingToScreen(string screenDeviceName)
 	{
 		if (CanMoveWindowAcrossScreens && TryGetBounds(screenDeviceName, out var bounds) && IsVisibleOnAnyScreen(bounds)) {
-			Form form = (Form)Control.FromHandle(Main.instance.Window.Handle);
+			Form form = Control.FromHandle(Main.instance.Window.Handle);
 			if (WouldViewFitInScreen(form.Bounds, bounds)) {
 				form.Location = new Point(bounds.Width / 2 - form.Width / 2 + bounds.X, bounds.Height / 2 - form.Height / 2 + bounds.Y);
 			}
