--- decompiled/TerrariaBuildable/SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj
+++ decompiled/TerrariaUnified/SteelSeriesEngineWrapper/SteelSeriesEngineWrapper.csproj
@@ -6,7 +_,7 @@
         <Version>1.0.0.0</Version>
         <Company>Copyright © 2020</Company>
         <RootNamespace>SteelSeriesEngineWrapper</RootNamespace>
-        <TargetFramework>net40</TargetFramework>
+        <TargetFramework>net8.0</TargetFramework>
         <LangVersion>latest</LangVersion>
     </PropertyGroup>
 
