--- decompiled/TerrariaBuildable/RailSDK/RailSDK.csproj
+++ decompiled/TerrariaUnified/RailSDK/RailSDK.csproj
@@ -1,12 +_,12 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
-    <Import Project="../Configuration.targets"/>
+    <Import Project="../Configuration.targets" />
 
     <PropertyGroup>
         <Version>1.0.0.0</Version>
         <Company>Copyright © 2018</Company>
         <RootNamespace>RailSDK.Net</RootNamespace>
-        <TargetFramework>net40</TargetFramework>
+        <TargetFramework>net6.0</TargetFramework>
         <LangVersion>latest</LangVersion>
     </PropertyGroup>
 
