--- decompiled/TerrariaClientWindows/Terraria/GameContent/UI/Chat/ItemTagHandler.cs
+++ decompiled/TerrariaClientLinux/Terraria/GameContent/UI/Chat/ItemTagHandler.cs
@@ -70,7 +_,7 @@
 		}
 		item.stack = 1;
 		if (options != null) {
-			string[] array = options.Split(',');
+			string[] array = options.Split(new char[1] { ',' });
 			for (int i = 0; i < array.Length; i++) {
 				if (array[i].Length == 0) {
 					continue;
