--- decompiled/TerrariaClientWindows/Terraria/UI/FancyErrorPrinter.cs
+++ decompiled/TerrariaClientLinux/Terraria/UI/FancyErrorPrinter.cs
@@ -114,7 +_,7 @@
 	private static void ShowTheBox(string preparedMessage)
 	{
 		if (Platform.IsWindows && !Main.dedServ) {
-			MessageBox.Show(preparedMessage, "Error", MessageBoxButtons.OK, MessageBoxIcon.Hand);
+			MessageBox.Show(preparedMessage, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
 		}
 	}
 }
