--- decompiled/TerrariaClientWindows/Terraria/UI/GameInterfaceLayer.cs
+++ decompiled/TerrariaClientLinux/Terraria/UI/GameInterfaceLayer.cs
@@ -19,19 +_,19 @@
 
 	public bool Draw()
 	{
-		Matrix transformMatrix;
+		Matrix transformationMatrix;
 		if (ScaleType == InterfaceScaleType.Game) {
 			PlayerInput.SetZoom_World();
-			transformMatrix = Main.GameViewMatrix.ZoomMatrix;
+			transformationMatrix = Main.GameViewMatrix.ZoomMatrix;
 		} else if (ScaleType == InterfaceScaleType.UI) {
 			PlayerInput.SetZoom_UI();
-			transformMatrix = Main.UIScaleMatrix;
+			transformationMatrix = Main.UIScaleMatrix;
 		} else {
 			PlayerInput.SetZoom_Unscaled();
-			transformMatrix = Matrix.Identity;
+			transformationMatrix = Matrix.Identity;
 		}
 		bool result = false;
-		Main.spriteBatch.Begin(SpriteSortMode.Deferred, null, null, null, null, null, transformMatrix);
+		Main.spriteBatch.Begin(SpriteSortMode.Deferred, null, null, null, null, null, transformationMatrix);
 		try {
 			result = DrawSelf();
 		} catch (Exception e) {
