--- decompiled/TerrariaServerWindows/Terraria/Audio/LegacyAudioSystem.cs
+++ decompiled/TerrariaServerMac/Terraria/Audio/LegacyAudioSystem.cs
@@ -30,7 +_,8 @@
 		List<IContentSource> fileSources = FileSources;
 		for (int i = 0; i < AudioTracks.Length; i++) {
 			if (TrackNamesByIndex.TryGetValue(i, out var value)) {
+				char directorySeparatorChar = Path.DirectorySeparatorChar;
-				string assetPath = "Music" + Path.DirectorySeparatorChar + value;
+				string assetPath = "Music" + directorySeparatorChar + value;
 				IAudioTrack audioTrack = DefaultTrackByIndex[i];
 				IAudioTrack audioTrack2 = audioTrack;
 				IAudioTrack audioTrack3 = FindReplacementTrack(fileSources, assetPath);
